src/main/java/com/scottlogic/sorts/KeywordPostSorter.java

```diff
@@ -0,0 +1,40 @@
+package com.scottlogic.sorts;
+
+import com.scottlogic.PostSorter;
+import com.scottlogic.SortOrder;
+import com.scottlogic.UserPost;
+import com.scottlogic.filters.KeywordPostFilter;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class KeywordPostSorter implements PostSorter {
+    private final String keyword;
+    private final KeywordPostFilter KeywordPostFilter;
+
+    public KeywordPostSorter(KeywordPostFilter KeywordPostFilter){
```

Comment:
This feels like overkill, you can probably just pass in the `keyword` and then make the `KeywordFilter` inside this class rather than being required to pass it in